[reland][AMD] Turn on TF32 for aten::mm

xw285cornell opened the pull request:
Summary:
hipblaslt supports TF32, so adding the support.

Original PR https://github.com/pytorch/pytorch/pull/139869

Test Plan: CI

Differential Revision: D67431681




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

xw285cornell commented:
@pytorchbot label "topic: not user facing"

jeffdaily commented on aten/src/ATen/Context.cpp:
```diff
@@ -237,10 +241,24 @@ void Context::setBenchmarkLimitCuDNN(int b) {
 
 void Context::setAllowTF32CuBLAS(bool b) {
+#ifdef USE_ROCM
+  const static auto allow_tf32 = c10::utils::check_env(hipblaslt_allow_tf32);
```
I understand why you want to make this env var static.  But the unit tests where you set the env var and then try to unset it later will have different behavior depending on how the UTs are sharded and the order in which they're run.  The first test to call allowTF32CuBLAS() will be sticky and you'll never be able to change the env var setting later.

jeffdaily replied:
Perhaps allowTF32CuBLAS is called infrequently enough to remove the static?  It's inside a USE_ROCM section so won't affect CUDA at all.

xw285cornell replied:
sounds good. I guess in the future after TF32 support is released in rocm, we can check the GPU arch here.

xw285cornell commented:
failed tests don't seem to be relevant to the changes here. Landing.

xw285cornell commented:
@pytorchbot merge -f "unrelated failures"

xw285cornell commented:
@jeffdaily I tried looking at those failed tests and they seem irrelevant (this PR shouldn't affect graph breaks etc). If it still breaks those tests please feel free to revert

malfet commented:
@pytorchbot revert -m "It breaks ROCM testing, see https://hud.pytorch.org/hud/pytorch/pytorch/06b4b96b34e93cc85f48b083cf04e7598fd68e20/1?per_page=50&name_filter=rocm%20%2F%20linux-focal-rocm6.2" -c ignoredsignal

xw285cornell committed:
[reland][AMD] Turn on TF32 for aten::mm (#143549)

Summary:

hipblaslt supports TF32, so adding the support.

Original PR https://github.com/pytorch/pytorch/pull/139869

Test Plan: CI

Reviewed By: leitian

Differential Revision: D67431681

jeanschmidt commented:
closing PR and re-exporting, as diff was landed internally but it is diverging from this PR.

jeanschmidt commented:
I tried to export and merge, but seems that this is not a viable approach. 

What is left now is to unland the internal diff and start again from a new diff, making sure to properly export.

I am taking note on why no signals on land time prevented the landing and forwarding to the owners.

clee2000 commented:
@pytorchbot revert -m "Seems to have broken some tests on rocm ex dynamo/test_misc.py::MiscTests::test_recompile_on_global_state_change [GH job link](https://github.com/pytorch/pytorch/actions/runs/12457661964/job/34774017859) [HUD commit link](https://hud.pytorch.org/pytorch/pytorch/commit/41cdc7f73552cc8a0dbf2d3cb55440c0d6b548ea).  It was flaky on this PR but fails consistently on the main branch, maybe a landrace?" -c landrace
